src/header/common/link.rs

```diff
@@ -0,0 +1,1027 @@
+    pub fn anchor(&self) -> Option<&String> {
+        self.anchor.as_ref()
+    }
+
+    /// Get the `LinkValue`'s `rev` parameter(s).
+    pub fn rev(&self) -> Option<&Vec<RelationType>> {
+        self.rev.as_ref()
+    }
+
+    /// Get the `LinkValue`'s `hreflang` parameter(s).
+    pub fn href_lang(&self) -> Option<&Vec<LanguageTag>> {
+        self.href_lang.as_ref()
+    }
+
+    /// Get the `LinkValue`'s `media` parameter(s).
+    pub fn media_desc(&self) -> Option<&Vec<MediaDesc>> {
+        self.media_desc.as_ref()
+    }
+
+    /// Get the `LinkValue`'s `title` parameter.
+    pub fn title(&self) -> Option<&String> {
+        self.title.as_ref()
+    }
+
+    /// Get the `LinkValue`'s `title*` parameter.
+    pub fn title_star(&self) -> Option<&String> {
+        self.title_star.as_ref()
+    }
+
+    /// Get the `LinkValue`'s `type` parameter.
+    pub fn media_type(&self) -> Option<&Mime> {
+        self.media_type.as_ref()
+    }
+
+    /// Get the `LinkValue`'s `link-extension` parameter.
+    pub fn link_extension(&self) -> Option<&String> {
+        self.link_extension.as_ref()
+    }
+
+    /// Add a `RelationType` to the `LinkValue`'s `rel` parameter.
+    pub fn push_rel(mut self, rel: RelationType) -> LinkValue {
+        let mut v = self.rel.take().unwrap_or(Vec::new());
+
+        v.push(rel);
+
+        self.rel = Some(v);
+
+        self
+    }
+
+    /// Set `LinkValue`'s `anchor` parameter.
+    pub fn set_anchor(mut self, anchor: &str) -> LinkValue {
+        self.anchor = Some(String::from(anchor));
+
+        self
+    }
+
+    /// Add a `RelationType` to the `LinkValue`'s `rev` parameter.
+    pub fn push_rev(mut self, rev: RelationType) -> LinkValue {
+        let mut v = self.rev.take().unwrap_or(Vec::new());
+
+        v.push(rev);
+
+        self.rev = Some(v);
+
+        self
+    }
+
+    /// Add a `LanguageTag` to the `LinkValue`'s `hreflang` parameter.
+    pub fn push_href_lang(mut self, language_tag: LanguageTag) -> LinkValue {
+        let mut v = self.href_lang.take().unwrap_or(Vec::new());
+
+        v.push(language_tag);
+
+        self.href_lang = Some(v);
+
+        self
+    }
+
+    /// Add a `MediaDesc` to the `LinkValue`'s `media_desc` parameter.
+    pub fn push_media_desc(mut self, media_desc: MediaDesc) -> LinkValue {
+        let mut v = self.media_desc.take().unwrap_or(Vec::new());
+
+        v.push(media_desc);
+
+        self.media_desc = Some(v);
+
+        self
+    }
+
+    /// Set `LinkValue`'s `title` parameter.
+    pub fn set_title(mut self, title: &str) -> LinkValue {
+        self.title = Some(String::from(title));
+
+        self
+    }
+
+    /// Set `LinkValue`'s `title*` parameter.
+    pub fn set_title_star(mut self, title_star: &str) -> LinkValue {
+        self.title_star = Some(String::from(title_star));
+
+        self
+    }
+
+    /// Set `LinkValue`'s `type` parameter.
+    pub fn set_media_type(mut self, media_type: &Mime) -> LinkValue {
```

Comment:
We should take the `Mime` by value, and allow the user to clone one or use the one they already have.